tardis/visualization/tools/interaction_radius_plot.py

```diff
@@ -0,0 +1,461 @@
+        previous_atomic_number = 0
+        for species_counter, identifier in enumerate(self.species):
+            if self._species_list is not None:
+                # Get the ion number and atomic number for each species
+                ion_number = identifier % 100
+                atomic_number = (identifier - ion_number) / 100
+                if previous_atomic_number == 0:
+                    # If this is the first species being plotted, then take note of the atomic number
+                    # don't update the colour index
+                    color_counter = color_counter
+                    previous_atomic_number = atomic_number
+                elif previous_atomic_number in self._keep_colour:
+                    # If the atomic number is in the list of elements that should all be plotted in the same colour
+                    # then don't update the colour index if this element has been plotted already
+                    if previous_atomic_number == atomic_number:
+                        color_counter = color_counter
+                        previous_atomic_number = atomic_number
+                    else:
+                        # Otherwise, increase the colour counter by one, because this is a new element
+                        color_counter = color_counter + 1
+                        previous_atomic_number = atomic_number
+                else:
+                    # If this is just a normal species that was requested then increment the colour index
+                    color_counter = color_counter + 1
+                    previous_atomic_number = atomic_number
+                # Calculate the colour of this species
+                color = self.cmap(color_counter / len(self._species_name))
+
+            else:
+                # If you're not using species list then this is just a fraction based on the total
+                # number of columns in the dataframe
+                color = self.cmap(species_counter / len(self.species))
+
+            color_list.append(color)
+
+        self._color_list = color_list
+
+        return
+
+    def generate_plot_mpl(
+        self,
+        packets_mode="virtual",
+        ax=None,
+        figsize=(11, 6),
+        cmapname="jet",
+        species_list=None,
+    ):
+        """
+        Generate the last interaction radius distribution plot using matplotlib.
+
+        Parameters
+        ----------
+        packets_mode : str, optional
+            Packet mode, either 'virtual' or 'real'. Default is 'virtual'.
+        ax : matplotlib.axes.Axes, optional
+            Axes object to plot on. If None, creates a new figure.
+        figsize : tuple, optional
+            Size of the figure. Default is (11, 6).
+        cmapname : str, optional
+            Colormap name. Default is 'jet'. A specific colormap can be chosen, such as "jet", "viridis", "plasma", etc.
+        species_list : list of str
+            List of species to plot. Default is None.
```

Comment:
There's no benefit of setting default here because you are raising ValueError for None, it should be a required parameter instead.